File: flink-datastream-1.7.1/src/main/scala/scalalang/timestamp/SensorTimeAssigner.scala
package scalalang.timestamp

import org.apache.flink.streaming.api.functions.timestamps.BoundedOutOfOrdernessTimestampExtractor
import org.apache.flink.streaming.api.windowing.time.Time

import scalalang.util.SensorReading

class SensorTimeAssigner extends BoundedOutOfOrdernessTimestampExtractor[SensorReading](Time.seconds(5)) {
  override def extractTimestamp(t: SensorReading): Long = t.timestamp
}

File: flink-datastream-1.11.2/src/main/java/datastream/timestamp/SensorTimeAssigner.java
package datastream.timestamp;

import datastream.util.SensorReading;
import org.apache.flink.streaming.api.functions.timestamps.BoundedOutOfOrdernessTimestampExtractor;
import org.apache.flink.streaming.api.windowing.time.Time;

public class SensorTimeAssigner extends BoundedOutOfOrdernessTimestampExtractor<SensorReading> {
    public SensorTimeAssigner(long maxDelayInSecs) {
        super(Time.seconds(maxDelayInSecs));
    }
    @Override
    public long extractTimestamp(SensorReading sensorReading) {
        return sensorReading.timestamp;
    }
}
